addons/l10n_ro_efactura/models/account_move.py

```diff
@@ -63,50 +66,74 @@ def _l10n_ro_edi_create_attachment_values(self, raw, res_model=None, res_id=None
         }
 
     def _l10n_ro_edi_create_document_invoice_sending(self, key_loading, attachment_raw):
-        """ Shorthand for creating a `l10n_ro_edi.document` of state `invoice_sending` """
+        # TODO in master: delete method and rename the next method as this one
+        pass
+
+    def _l10n_ro_edi_create_document_invoice_sending_from_values(self, values: dict):
+        """ Shorthand for creating a ``l10n_ro_edi.document`` of state ``invoice_sending``.
+
+        :param values: dictionary of {'key_loading': <str>, 'attachment_raw': <bytes>}
+        :return: ``l10n_ro_edi.document`` object """
         self.ensure_one()
         document = self.env['l10n_ro_edi.document'].sudo().create({
             'invoice_id': self.id,
             'state': 'invoice_sending',
-            'key_loading': key_loading,
+            'key_loading': values['key_loading'],
         })
         attachment_values = self._l10n_ro_edi_create_attachment_values(
-            raw=attachment_raw,
+            raw=values['attachment_raw'],
             res_model=document._name,
             res_id=document.id,
         )
         document.attachment_id = self.env['ir.attachment'].sudo().create(attachment_values)
         return document
 
     def _l10n_ro_edi_create_document_invoice_sending_failed(self, message, attachment_raw=None):
+        # TODO in master: delete method and rename the next method as this one
+        pass
+
+    def _l10n_ro_edi_create_document_invoice_sending_failed_from_values(self, values: dict):
```

Comment:
Renaming the method seems useless. The only parameter you add is `key_loading` that is optional so you could just add it as it is:
`    def _l10n_ro_edi_create_document_invoice_sending_failed(self, message, attachment_raw=None, key_loading=None):`

Reply:
My original goal was to make it general for all the create document functions, but I guess I'll just do it in master :)

(ready for review :+1:)